library/aes.c

```diff
@@ -971,6 +971,7 @@ int mbedtls_aes_crypt_ecb( mbedtls_aes_context *ctx,
                            unsigned char output[16] )
 {
     AES_VALIDATE_RET( ctx != NULL );
+    AES_VALIDATE_RET( ctx->rk != NULL );
```

Comment:
All the `VALIDATE` macros are no-op since 3.0. They're part of a discontinued feature.

Furthermore, the intent of this feature is to return an error when a user calls a library function with a null pointer. `ctx->rk` is not a parameter and cannot be null unless a user calls a library function out of sequence (crypt without setkey) which the low-level modules don't try to protect against. (I personally think it would make more sense to protect against that than against null pointer, but we shouldn't bother to do it in low-level modules that users shouldn't call directly, and anyway it wasn't an objective of CHECK_PARAMS.)